### PR TITLE
Change docker base images from debian to ubuntu

### LINKED ISSUE #3045: Fix docker scan vulnerabilities
Checking the [Dockerfile](https://github.com/hyperledger/besu/blob/master/docker/openjdk-11/Dockerfile) seems that base image `openjdk:11.0.7-jre-slim-buster` is bringing some vulnerabilities flagged by some image scanning tools like snyk or harbor.

To get list of vulnerabilities you can run following command:

```
 docker scan -f docker/openjdk-11/Dockerfile hyperledger/besu:develop-openjdk-11 --severity high
```

Producing following output:
```
Tested 91 dependencies for known issues, found 10 issues.

Base Image                  Vulnerabilities  Severity
openjdk:11-jre-slim-buster  62               2 critical, 8 high, 6 medium, 46 low
```

In addition some alternative base images are recommended:

```
Recommendations for base image upgrade:

Alternative image types
Base Image                         Vulnerabilities  Severity
openjdk:17-ea-22-jdk-oracle        0                0 critical, 0 high, 0 medium, 0 low
openjdk:16-ea-33                   0                0 critical, 0 high, 0 medium, 0 low
openjdk:17-ea-10-jdk-oracle        0                0 critical, 0 high, 0 medium, 0 low
openjdk:17-ea-26-jdk-oraclelinux8  0                0 critical, 0 high, 0 medium, 0 low
```

It is not critical but some private docker registry won't allow to pull such images with high severity vulnerabilities.

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Change the docker base images from debian (implied by openjdk image) to
ubuntu explicitly. Ubuntu has quicker reaction time to base image
security vulnerabilities.

## Fixed Issue(s)

Fixes #3045

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,20 +2,19 @@
...
+- Represent baseFee as Wei instead of long accordingly to the spec [#2785](https://github.com/hyperledger/besu/issues/2785)
 - Adding support of the NO_COLOR environment variable as described in the [NO_COLOR](https://no-color.org/) standard [#3085](https://github.com/hyperledger/besu/pull/3085)
 - Add `privx_findFlexiblePrivacyGroup` RPC Method, `privx_findOnchainPrivacyGroup` will be removed in a future release [#3075](https://github.com/hyperledger/besu/pull/3075)
 - The invalid value is now shown when `--bootnodes` cannot parse an item to make it easier to identify which option is invalid. 
 - Adding two new options to be able to specify desired TLS protocol version and Java cipher suites [#3105](https://github.com/hyperledger/besu/pull/3105)
 - Implements [EIP-4399](https://eips.ethereum.org/EIPS/eip-4399) to repurpose DIFFICULTY opcode after the merge as a source of entropy from the Beacon chain. [#3081](https://github.com/hyperledger/besu/issues/3081)
 
 ### Bug Fixes
+- Change the base docker image from Debian Buster to Ubuntu 20.04 [#3171](https://github.com/hyperledger/besu/issues/3171) fixes [#3045](https://github.com/hyperledger/besu/issues/3171)
```
Comment (macfarla) [inline CHANGELOG.md:17]: fixes [#3045](https://github.com/hyperledger/besu/issues/3045)

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Review by antonydenyer — APPROVED** [review]

**Comment by jmcruz1983** [comment]
Any idea when this improvement would be publicly released?

**Comment by shemnon** [comment]
We are going into a quarterly release cycle, so it will hit as RC and the next "full" release will be in Jan.  if you would like a 21.10.5 release feel free to ask in https://chat.hyperledger.org/channel/besu-contributors

**Comment by jmcruz1983** [comment]
> We are going into a quarterly release cycle, so it will hit as RC and the next "full" release will be in Jan. if you would like a 21.10.4 release feel free to ask in https://chat.hyperledger.org/channel/besu-contributors

Thanks @shemnon I am ok with using RC until next full release arrives.
What is the RC version?

**Comment by macfarla** [comment]
@jmcruz1983 you can either go with the RC https://github.com/hyperledger/besu/releases/tag/22.1.0-RC1, 
OR we included this update in 21.10.5 (which also includes the latest log4j release) https://github.com/hyperledger/besu/releases/tag/21.10.5

**Comment by jmcruz1983** [comment]
Thanks @macfarla 
just FYI, 
I was doing some last scanning to both  RC & 21.10.5 images and I am getting following critical vulnerability:
[GHSA-94g7-hpv8-h9qm](https://www.iboxshare.com/advisories/GHSA-94g7-hpv8-h9qm)

```
trivy image --severity CRITICAL hyperledger/besu:21.10.5             
2021-12-20T10:50:12.988+0100    INFO    Detected OS: ubuntu
2021-12-20T10:50:12.988+0100    INFO    Detecting Ubuntu vulnerabilities...
2021-12-20T10:50:12.990+0100    INFO    Number of language-specific files: 209
2021-12-20T10:50:12.990+0100    INFO    Detecting jar vulnerabilities...

hyperledger/besu:21.10.5 (ubuntu 20.04)
=======================================
Total: 0 (CRITICAL: 0)


opt/besu/lib/splunk-library-javalogging-1.9.0.jar (jar)
=======================================================
Total: 1 (CRITICAL: 1)

+-----------------------------------------------+---------------------+----------+-------------------+---------------+----------------------------------------------+
|                    LIBRARY                    |  VULNERABILITY ID   | SEVERITY | INSTALLED VERSION | FIXED VERSION |                    TITLE                     |
+-----------------------------------------------+---------------------+----------+-------------------+---------------+----------------------------------------------+
| com.splunk.logging:splunk-library-javalogging | GHSA-94g7-hpv8-h9qm | CRITICAL | 1.9.0             | 1.11.1        | Remote code injection in Log4j               |
|                                               |                     |          |                   |               | -->github.com/advisories/GHSA-94g7-hpv8-h9qm |
+-----------------------------------------------+---------------------+----------+-------------------+---------------+----------------------------------------------+
```

**Comment by shemnon** [comment]
The scanner is incorrect.  

If you read the vulnerability (https://github.com/splunk/splunk-library-javalogging/security/advisories/GHSA-94g7-hpv8-h9qm) what splunk is doing is changing their transitive dependency on log4j to 2.16.0 (which isn't even the latest!).  The CVE doesn't apply to `splunk-library-javalogging-1.9.0.jar` but to the log4j jars a maven dependency would bring in.  There is no changes to the code in the jars.

So this scan result is, in my opinion, a false positive.  Via a dependency plugin we fix th elog4j version to 2.17.0.

But @atoulme should probably look into getting us onto the latest splunk jars anyway.

**Comment by atoulme** [comment]
Thanks for the prod. I will follow up.

**Pull request merged**